src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -437,9 +437,12 @@ private function getHiddenResponse(OutputInterface $output, $inputStream, bool $
             throw new RuntimeException('Unable to hide the response.');
         }
 
-        $inputHelper?->waitForInput();
-
-        $value = fgets($inputStream, 4096);
+        if (null !== $inputHelper) {
+            $inputHelper->waitForInput();
+            $value = fgets($inputStream, 4096);
+        } else {
+            $value = $this->doReadInput($inputStream);
+        }
 
         if (4095 === \strlen($value)) {
```

Comment:
would it make sense to move this check to the doReadInput method? i'm wondering why we have it only here and not next to the other calls to fgets